app/src/main/java/org/greenstand/android/TreeTracker/viewmodels/MapViewModel.kt

```diff
@@ -33,6 +35,10 @@ class MapViewModel constructor(private val validateCheckInStatusUseCase: Validat
         }
     }
 
+    suspend fun requiresLogin(): Boolean {
```

Comment:
I had to add this function since I don't see a good way to use the existing boolean MutableLiveData in a `Add tree button` click listener. Let me know if there is a clean and simple way to use it, I will change it. 

For now, this looks like the straight forward way to implement this change.